### PR TITLE
Replace deprecated .get_pointer_to_global() 

### PR DESCRIPTION
**Pull request by sklam**
... with .get_function_address() and .get_global_value_address()

### CODE REVIEW

**Comment by seibert** [comment]
This looks good to me.  @pitrou?

File: llvmlite/binding/executionengine.py [hunk 1]
```diff
@@ -53,15 +55,25 @@ def __init__(self, ptr, module):
...
-        # getGlobalValueAddress should be used instead.
+    def get_pointer_to_function(self, gv):
+        warnings.warn(".get_pointer_to_function() is deprecated.  Use "
+                      ".get_function_address() instead.", DeprecationWarning)
         ptr = ffi.lib.LLVMPY_GetPointerToGlobal(self, gv)
         if ptr is None:
             raise ValueError("Cannot find given global value %r" % (gv.name))
         return ptr
 
-    get_pointer_to_function = get_pointer_to_global
```
Comment (pitrou) [inline llvmlite/binding/executionengine.py]: It looks a bit weird to deprecate one function but remove the other.

Comment (sklam) [inline llvmlite/binding/executionengine.py]: The name ``get_pointer_to_global()``  implies it works on any global values but it does not.  It gives incorrect result for non-function global values.  With ``get_pointer_to_function()``, the name implies that it is expecting a function value and it works correctly.  I intentionally break the API to help user avoid getting a segfault.

Comment (sklam) [inline llvmlite/binding/executionengine.py]: Also note that ``get_pointer_to_global()`` on global values returns a pointer within the address range of the emitted object module.  One can write to the returned value but who knows what memory is being corrupted.

Comment (pitrou) [inline llvmlite/binding/executionengine.py]: Thanks for the explanation. Could you update the docs? `get_pointer_to_global` is mentioned there.

**Pull request merged**